### PR TITLE
Add bandwidth api as a sms / mms alternitive

### PR DESCRIPTION
**Pull request by Jmainguy**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

bandwidth
##### ANSIBLE VERSION

2.3
##### SUMMARY

Adds the bandwidth https://catapult.inetwork.com/ api as an option for sending sms / mms notifications from ansible.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @Jmainguy for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ryansb** [comment]
Please check the TravisCI build logs and correct the module docs - they're not formatted right.

**Comment by Jmainguy** [comment]
Huzzah it passes now, @ryansb Can you give it another look over now?

**Comment by Jmainguy** [comment]
@resmo nice catch, I have added no_log=True for all auth stuff now. Can you take another look?

File: notification/bandwidth.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  api_secret: 
+    description:
+      - Api Secret from Api account page
+    required: true
+    default: null
+
+author: "Jonathan Mainguy (@Jmainguy)"
+notes:
+    - Will return changed even if the media url is wrong
+    - Will return changed if the destination number is invalid
```
Comment (ryansb) [inline notification/bandwidth.py:68]: Should this be a `fail` instead? Seems like silently letting numbers be invalid could lead to issues.

**Comment by ryansb** [comment]
Code looks good, just want to check whether wrong numbers should be `changed` or `failed`.

**Comment by Jmainguy** [comment]
@ryansb You are correct, it is confusing, and not ideal. However the API just responds with 201 whether its a good number, or a bad number, and its just not supported at this time (by the bandwidth catapult api). I have asked them to consider adding this feature in the future so we can utilize it.

That being said, since the API does not support it, there is not anything I can do to add it here at this time.

Any other good questions like the above from anybody else?

**Comment by Jmainguy** [comment]
Reopened in new repo. https://github.com/ansible/ansible/pull/23052

**Pull request closed without merging**